src/live.rs

```diff
@@ -343,6 +350,52 @@ impl<'a> KargEmbedAreas<'a> {
         get_kargs_at_offset(self.file, self.length, self.default_kargs_offset)
     }
 
+    fn format_kargs(&mut self, kargs: &str) -> Result<Vec<u8>> {
+        let mut new_area = vec![b'#'; self.length];
+        new_area[..kargs.len()].copy_from_slice(kargs.as_bytes());
+        Ok(new_area)
+    }
+
+    fn stream(&mut self, kargs: &str, writer: &mut (impl Write + ?Sized)) -> Result<()> {
+        let mut buf = [0u8; BUFFER_SIZE];
+        let kargs: String = kargs.trim().to_string() + "\n";
+        if kargs.len() > self.length {
+            bail!(
+                "kargs too large for area: {} vs {}",
+                kargs.len(),
+                self.length
+            );
+        }
+
+        self.file
+            .seek(SeekFrom::Start(0))
+            .context("seeking to start")?;
+        let new_area = self.format_kargs(&kargs).unwrap();
```

Comment:
In this codebase, the hierarchy of error handling is:

1. `unwrap()` - obvious programmer errors, like extracting a `Some()` value when we just checked that we're not in the `None` case
2. `expect("explanation")` - less obvious programmer errors, like an inconsistent state caused by another method, where an `unwrap()` won't be sufficiently clear for debugging
3. `?` - legitimate runtime errors that should be propagated to the caller, but we don't need to add additional context to the error produced by the callee (e.g. the callee is a helper function)
4. `context("explanation")?` or `with_context(|| format!("explanation for {}", something))?` - legitimate runtime errors where we know something the callee doesn't

I think this is case 3.